packages/dashboard-core-plugins/src/panels/DropdownFilterPanel.tsx

```diff
@@ -160,25 +225,43 @@ class DropdownFilterPanel extends Component {
     if (this.cleanup) {
       this.cleanup();
     }
-    if (valuesTable !== null) {
+    if (valuesTable != null) {
       valuesTable.close();
     }
   }
 
-  getCachedValues = memoize((rawValues, { type, name }, formatter) => {
-    if (TableUtils.isDateType(type)) {
+  dropdownFilterRef: RefObject<DropdownFilter> | null;
+
+  panelContainer: RefObject<HTMLDivElement> | null;
+
+  pending: Pending;
+
+  cleanup?: () => void;
+
+  columnFormats?: FormattingRule[];
+
+  getCachedValues = memoize(
+    (
+      rawValues: unknown[],
+      { type, name }: DropdownFilterColumn,
+      formatter: Formatter
+    ) => {
+      if (TableUtils.isDateType(type)) {
+        return rawValues.map(value =>
+          DropdownFilterPanel.DATETIME_FORMATTER.format(value as number)
+        );
+      }
       return rawValues.map(value =>
-        DropdownFilterPanel.DATETIME_FORMATTER.format(value)
+        // Skip formatting for nulls so they don't get converted to ''
+        value != null && type != null
+          ? formatter.getFormattedString(value, type, name)
+          : null
       );
     }
-    return rawValues.map(value =>
-      // Skip formatting for nulls so they don't get converted to ''
-      value != null ? formatter.getFormattedString(value, type, name) : null
-    );
-  });
+  );
 
-  getCoordinateForColumn() {
-    if (!this.panelContainer.current) {
+  getCoordinateForColumn(): [number, number] | null {
+    if (!this.panelContainer?.current) {
```

Comment:
`this.panelContainer` is never null, no need to do an optional check.
That being said, we should be more explicit about these checks instead of relying on the truthy/falsy value anyway:
```suggestion
    if (this.panelContainer.current == null) {
```